src/test/java/org/cactoos/io/ZipTest.java

```diff
@@ -24,51 +24,80 @@
 
 package org.cactoos.io;
 
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
-import org.hamcrest.MatcherAssert;
-import org.hamcrest.Matchers;
+import java.io.File;
+import java.io.OutputStream;
+import java.util.zip.ZipFile;
+import org.cactoos.Text;
+import org.cactoos.iterator.Mapped;
+import org.cactoos.list.ListOf;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.text.Joined;
+import org.cactoos.text.TextOf;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.ScalarHasValue;
 
 /**
  * Test case for {@link Zip}.
  *
  * @since 0.29
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public final class ZipTest {
+
     /**
      * Temporary folder.
      */
     @Rule
-    public final TemporaryFolder folder = new TemporaryFolder();
+    public final TemporaryFolder temporal = new TemporaryFolder();
 
     @Test
-    public void zip() throws Exception {
-        final Path dir = this.folder.newFolder().toPath();
-        dir.resolve("x/y").toFile().mkdirs();
-        Files.write(dir.resolve("x/y/test"), "".getBytes());
-        try (ZipInputStream input = new ZipInputStream(
-            new Zip(
-                new Directory(dir)
-            ).stream()
-        )) {
-            int cnt = 0;
-            ZipEntry entry = input.getNextEntry();
-            while (entry != null) {
-                ++cnt;
-                entry = input.getNextEntry();
-            }
-            MatcherAssert.assertThat(
-                "Can't list files in a directory represented by a path",
-                cnt,
-                // @checkstyle MagicNumber (1 line)
-                Matchers.equalTo(4)
-            );
-        }
+    public void mustZipDirectory() throws Exception {
+        final String zipname = "abc.zip";
+        final File folder = this.temporal.newFolder("abc");
+        new File(folder, "A.txt").createNewFile();
+        new File(folder, "B").mkdir();
+        new File(
+            new File(
+                new Joined(File.separator, folder.getPath(), "B").asString()
+            ), "B.txt"
```

Comment:
@fanifieiev this seems very complex, what is wrong with: `new File(new File(folder, "B"), "B.txt"))` ? You can even allocate `new File(folder, "B")` to a variable to be able to not repeat yourself

Reply:
@victornoel Agree